lib/src/signed_video_h26x_common.c

```diff
@@ -868,7 +923,13 @@ check_and_copy_hash_to_hash_list(signed_video_t *self, const uint8_t *hash, size
   uint8_t *hash_list = &self->gop_info->hash_list[0];
   int *list_idx = &self->gop_info->list_idx;
   // Check if there is room for another hash in the |hash_list|.
-  if (*list_idx + hash_size > self->gop_info->hash_list_size) *list_idx = -1;
+  if (*list_idx + hash_size > self->gop_info->hash_list_size) {
+    init_fallback_gop_hash(self, hash_list, *list_idx);
```

Comment:
First of all, the function has a return code that you should check. Secondly, the function only has two operations which could be done here directly, hence no need for a function. Thirdly, switching to a different kind of operation in the middle of the gop hash computation process makes the code more vulnerable to errors and introduction of bugs. May I suggest to use the "secondary ctx" from the start?